wakuv2/waku.go

```diff
@@ -1014,14 +1101,87 @@ func (w *Waku) Send(pubsubTopic string, msg *pb.WakuMessage) ([]byte, error) {
 	alreadyCached := w.envelopeCache.Has(gethcommon.BytesToHash(envelope.Hash().Bytes()))
 	w.poolMu.Unlock()
 	if !alreadyCached {
-		recvMessage := common.NewReceivedMessage(envelope, common.RelayedMessageType)
+		recvMessage := common.NewReceivedMessage(envelope, common.SendMessageType)
 		w.postEvent(recvMessage) // notify the local node about the new message
 		w.addEnvelope(recvMessage)
 	}
 
 	return envelope.Hash().Bytes(), nil
 }
 
+// ctx, peer, r.PubsubTopic, contentTopics, uint64(r.From), uint64(r.To), options, processEnvelopes
+func (w *Waku) messageHashBasedQuery(ctx context.Context, hashes []gethcommon.Hash, pubsubTopic string) []gethcommon.Hash {
+	selectedPeer := w.storePeerID
+	if selectedPeer == "" {
+		selectedPeers, err := w.node.PeerManager().SelectPeers(
+			peermanager.PeerSelectionCriteria{
+				SelectionType: peermanager.Automatic,
+				Proto:         store.StoreQueryID_v300,
+				PubsubTopics:  []string{pubsubTopic},
+				Ctx:           ctx,
+			},
+		)
+		if err != nil {
+			w.logger.Error("could not select peers", zap.Error(err))
+			return []gethcommon.Hash{}
+		}
+		selectedPeer = selectedPeers[0]
+	}
```

Comment:
I wonder if we should have an automatic peer selection here, since for now we have a fixed list of store nodes for status :thinking: